dnsapi/dns_1984hosting.sh

```diff
@@ -150,7 +150,7 @@ _1984hosting_login() {
   _debug2 response "$response"
 
   if _contains "$response" '"loggedin": true'; then
-    One984HOSTING_COOKIE="$(grep -i '^set-cookie:' "$HTTP_HEADER" | _tail_n 1 | _egrep_o 'sessionid=[^;]*;' | tr -d ';')"
+    One984HOSTING_COOKIE="$(grep -i '^set-cookie:' "$HTTP_HEADER" | _head_n 1 | _egrep_o 'sessionid=[^;]*;' | tr -d ';')"
```

Comment:
why not grep for "sessionid" instead?

Reply:
I got preoccupied with fixing the bug while preserving how the script currently behaves. While my change currently fixes things, removing the unnecessary step of using head/tail and grepping directly for the `sessionid` is more "future-proof" if the `set-cookie` lines change order and/or increase/decrease again. To go a step further, I used cat on a test cookie text file using the example I posted above to see if removing the first grep does not break anything, since it seems just as unnecessary as I'm looking at it again. It seems to work fine from typing it into the terminal. I'm going to test it with the script and commit the change it if it works.

Reply:
> I got preoccupied with fixing the bug while preserving how the script currently behaves. While my change currently fixes things, removing the unnecessary step of using head/tail and grepping directly for the `sessionid` is more "future-proof" if the `set-cookie` lines change order and/or increase/decrease again. To go a step further, I used cat on a test cookie text file using the example I posted above to see if removing the first grep does not break anything, since it seems just as unnecessary as I'm looking at it again. It seems to work fine from typing it into the terminal. I'm going to test it with the script and commit the change it if it works.

So I must need coffee or something, since that first grep is what is getting the HTTP header piped into egrep and tr in the first place. It is still possible to slim it down to one grep/egrep though since we can skip straight to the `sessionid` and not bother with the `set-cookie:` pattern.

Reply:
I committed the simplified grepping. I tested it and it works just fine for me on pfsense.